assets/src/blocks/amp-story/edit.js

```diff
@@ -74,33 +78,175 @@ export default class EditPage extends Component {
+			mediaUrl: media.url,
+			mediaId: media.id,
+			mediaType: mediaType,
+		} );
+
+		if ( IMAGE_BACKGROUND_TYPE === mediaType ) {
+			this.props.setAttributes( { poster: undefined } );
+		}
 	}
 
 	render() {
-		const props = this.props;
-		const { attributes } = props;
+		const { attributes, media, setAttributes } = this.props;
+
+		const { backgroundColor, mediaId, mediaType, mediaUrl, focalPoint, poster } = attributes;
+
+		const instructions = <p>{ __( 'To edit the background image or video, you need permission to upload media.', 'amp' ) }</p>;
+
+		const style = {
+			backgroundColor: backgroundColor,
+			backgroundImage: IMAGE_BACKGROUND_TYPE === mediaType && mediaUrl ? `url(${ mediaUrl })` : undefined,
+			backgroundPosition: IMAGE_BACKGROUND_TYPE === mediaType && focalPoint ? `${ focalPoint.x * 100 }% ${ focalPoint.y * 100 }%` : 'cover',
+			backgroundRepeat: 'no-repeat',
+		};
+
+		if ( VIDEO_BACKGROUND_TYPE === mediaType && poster ) {
+			style.backgroundImage = `url(${ poster })`;
+		}
 
 		return (
 			<Fragment>
 				<InspectorControls key="controls">
 					<PanelColorSettings
-						title={ __( 'Background Color Settings', 'amp' ) }
+						title={ __( 'Color Settings', 'amp' ) }
 						initialOpen={ false }
 						colorSettings={ [
 							{
-								value: attributes.backgroundColor,
-								onChange: this.onChangeBackgroundColor,
+								value: backgroundColor,
+								onChange: ( value ) => setAttributes( { backgroundColor: value } ),
 								label: __( 'Background Color', 'amp' ),
 							},
 						] }
 					/>
+					<PanelBody title={ __( 'Background Media', 'amp' ) }>
+						<Fragment>
+							<BaseControl>
+								<MediaUploadCheck fallback={ instructions }>
+									<MediaUpload
+										onSelect={ this.onSelectMedia }
+										allowedTypes={ ALLOWED_MEDIA_TYPES }
+										value={ mediaId }
+										render={ ( { open } ) => (
+											<Button isDefault isLarge onClick={ open } className={ 'editor-amp-story-page-background' }>
+												{ mediaUrl ? __( 'Edit Media', 'amp' ) : __( 'Upload Media', 'amp' ) }
+											</Button>
+										) }
+									/>
+								</MediaUploadCheck>
+								{ !! mediaId &&
+								<MediaUploadCheck>
+									<Button onClick={ () => setAttributes( { mediaUrl: undefined, mediaId: undefined, mediaType: undefined } ) } isLink isDestructive>
+										{ VIDEO_BACKGROUND_TYPE === mediaType ? __( 'Remove video', 'amp' ) : __( 'Remove image', 'amp' ) }
+									</Button>
+								</MediaUploadCheck>
+								}
+							</BaseControl>
+							{ VIDEO_BACKGROUND_TYPE === mediaType && (
+								<MediaUploadCheck>
+									<BaseControl
+										label={ __( 'Poster Image', 'amp' ) }
+									>
+										<MediaUpload
+											title={ __( 'Select Poster Image', 'amp' ) }
+											onSelect={ ( image ) => setAttributes( { poster: image.url } ) }
+											allowedTypes={ POSTER_ALLOWED_MEDIA_TYPES }
+											render={ ( { open } ) => (
+												<Button
```

Comment:
Could we make this required? Or at least let the user know that this is mandatory -- the video wouldn't show up otherwise on the AMP Story (would get sanitized by the plugin's logic). Thoughts?

Reply:
Changed the label:

<img width="286" alt="screenshot 2019-02-28 at 17 10 17" src="https://user-images.githubusercontent.com/841956/53580388-c8f49880-3b7b-11e9-8cce-677ab2d5134f.png">